eng/common/scripts/Package-Properties.ps1

```diff
@@ -120,38 +109,76 @@ function Get-AllPkgProperties ([string]$ServiceDirectory = $null)
         foreach ($dir in (Get-ChildItem $searchDir -Directory))
         {
             $serviceDir = Join-Path $searchDir $dir.Name
+            $ciFiles = fetchCiFiles -serviceDirectory $serviceDir -includeincludeMgmtPkg $includeMgmt
 
-            if (Test-Path (Join-Path $serviceDir "ci.yml"))
+            foreach ($ciFile in $ciFiles)
             {
-                $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
-                if ($activePkgList -ne $null)
-                {
-                    $pkgPropsResult = Operate-OnPackages -activePkgList $activePkgList -ServiceDirectory $dir.Name -pkgPropsResult $pkgPropsResult
-                }
+                $pkgPropsResult = Operate-OnPackages -ciFilePath $ciFile -ServiceDirectory $dir.Name -pkgPropsResult $pkgPropsResult
             }
         }
     } 
     else
     {
         $serviceDir = Join-Path $RepoRoot "sdk" $ServiceDirectory
-        if (Test-Path (Join-Path $serviceDir "ci.yml"))
-        {
-            $activePkgList = Get-PkgListFromYml -ciYmlPath (Join-Path $serviceDir "ci.yml")
-            if ($activePkgList -ne $null)
-            {
-                $pkgPropsResult = Operate-OnPackages -activePkgList $activePkgList -ServiceDirectory $ServiceDirectory -pkgPropsResult $pkgPropsResult
-            }
+        $ciFiles = fetchCiFiles -serviceDirectory $serviceDir -includeincludeMgmtPkg $includeMgmt
+        foreach ($ciFile in $ciFiles) {
+            $pkgPropsResult = Operate-OnPackages -ciFilePath $ciFile -ServiceDirectory $ServiceDirectory -pkgPropsResult $pkgPropsResult
         }
     }
 
     return $pkgPropsResult
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function GetMetaData($lang){
+    switch ($lang) {
+      "java" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/java-packages.csv"
+        break
+      }
+      ".net" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/dotnet-packages.csv"
+        break
+      }
+      "python" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/python-packages.csv"
+        break
+      }
+      "javascript" {
+        $metadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/js-packages.csv"
+        break
+      }
+      default {
+        Write-Host "Unrecognized Language: $language"
+        exit(1)
+      }
+    }
+  
+    $metadataResponse = Invoke-RestMethod -Uri $metadataUri -method "GET" -MaximumRetryCount 3 -RetryIntervalSec 10 | ConvertFrom-Csv
+  
+    return $metadataResponse
+}
+
+function fetchCiFiles([string]$serviceDirectory, [bool]$includeMgmtPkg) {
```

Comment:
Is there a reason you need this instead of just doing a wildcard `ci*.yml`?

Reply:
Explained above